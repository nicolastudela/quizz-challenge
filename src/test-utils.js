/* eslint-disable react/prop-types */
import React from "react";
import { ThemeProvider } from "@material-ui/core/styles";
import { MemoryRouter } from "react-router-dom";
import { Provider } from "react-redux";
import thunk from "redux-thunk";
import configureStore from "redux-mock-store";
import theme from "./theme";

export const INITIAL_EMPTY_STORE = {
  user: {},
  questionnaire: {
    items: [],
    isFetching: false,
    error: null
  }
};

const mockStore = configureStore([thunk]);

export const BasicTestWrapper = ({ children }) => {
  return (
    <ThemeProvider theme={theme}>
      <MemoryRouter>{children}</MemoryRouter>
    </ThemeProvider>
  );
};

export const WithReduxStoreTestWrapper = ({
  children,
  store = INITIAL_EMPTY_STORE,
  initialEntries,
  initialIndex
}) => {
  return (
    <ThemeProvider theme={theme}>
      <MemoryRouter initialEntries={initialEntries} initialIndex={initialIndex}>
        <Provider store={mockStore(store)}>{children}</Provider>
      </MemoryRouter>
    </ThemeProvider>
  );
};
