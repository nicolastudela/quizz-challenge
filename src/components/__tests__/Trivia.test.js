import React from "react";
import renderer from "react-test-renderer";
import { render, wait } from "@testing-library/react";
import { WithReduxStoreTestWrapper, BasicTestWrapper } from "test-utils";
import ConnectedTrivia, { Trivia } from "../Trivia";
import mockQuestions from "../../__mocks__/booleanQuestionnaireResponse.mock";

jest.mock("@material-ui/core", () => {
  const original = jest.requireActual("@material-ui/core");
  return {
    ...original,
    Grow: jest.fn().mockImplementation(props => <div>{props.children}</div>)
  };
});

describe("Triva page", () => {
  it("renders propery ", () => {
    expect.assertions(1);
    const component = renderer.create(
      <BasicTestWrapper>
        <Trivia
          userName="Nico"
          isQuestionnaireLoading={false}
          questions={mockQuestions.results}
        />
      </BasicTestWrapper>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("will show mocked questions when it loads ", () => {
    expect.assertions(2);
    const { getByText } = render(
      <WithReduxStoreTestWrapper
        store={{
          questionnaire: {
            items: mockQuestions.results,
            isFetching: false,
            error: null
          },
          user: { name: "Nico" }
        }}
      >
        <ConnectedTrivia userName="Nico" />
      </WithReduxStoreTestWrapper>
    );

    expect(getByText(mockQuestions.results[0].category)).toBeTruthy();
    expect(getByText(mockQuestions.results[0].question)).toBeTruthy();
  });

  it("will show the second question after answering the first one ", async () => {
    expect.assertions(4);
    const { getByText } = render(
      <WithReduxStoreTestWrapper
        store={{
          questionnaire: {
            items: mockQuestions.results,
            isFetching: false,
            error: null
          },
          user: { name: "Nico" }
        }}
      >
        <ConnectedTrivia userName="Nico" />
      </WithReduxStoreTestWrapper>
    );

    expect(getByText("1 / 10")).toBeTruthy();

    // asnwers True
    const answerButtonTrue = getByText("True");
    answerButtonTrue.click();

    await wait(() => expect(getByText("2 / 10")).toBeTruthy());

    expect(getByText(mockQuestions.results[1].category)).toBeTruthy();
    expect(getByText(mockQuestions.results[1].question)).toBeTruthy();
  });
});
