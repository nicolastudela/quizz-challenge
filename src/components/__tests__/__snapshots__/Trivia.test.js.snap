// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Triva page renders propery  1`] = `
Array [
  <h3
    className="MuiTypography-root MuiTypography-h3 MuiTypography-alignCenter"
  >
    Go Nico!
  </h3>,
  <div
    className="MuiBox-root MuiBox-root-32"
  />,
  <div
    className="MuiBox-root MuiBox-root-33"
  >
    <div>
      <div>
        <h4
          className="MuiTypography-root MuiTypography-h4 makeStyles-h4-34 MuiTypography-alignCenter"
        >
          History
        </h4>
        <div
          className="MuiBox-root MuiBox-root-35"
        />
        <h5
          className="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter"
        >
          The Kingdom of Prussia briefly held land in Estonia.
        </h5>
        <div
          className="MuiBox-root MuiBox-root-36"
        />
        <div
          className="MuiBox-root MuiBox-root-37"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-containedSizeLarge MuiButton-sizeLarge"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "width": "30%",
              }
            }
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              False
            </span>
          </button>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "width": "30%",
              }
            }
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              True
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      className="MuiBox-root MuiBox-root-69"
    />
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      1 / 10
    </p>
  </div>,
]
`;
